compiler/rustc_parse/src/lexer/unescape_error_reporting.rs

```diff
@@ -199,16 +199,7 @@ pub(crate) fn emit_unescape_error(
             err.span_label(span, format!("must be ASCII{postfix}"));
             // Note: the \\xHH suggestions are not given for raw byte string
             // literals, because they are araw and so cannot use any escapes.
-            if (c as u32) <= 0xFF && mode != Mode::RawByteStr {
-                err.span_suggestion(
-                    span,
-                    format!(
-                        "if you meant to use the unicode code point for {c:?}, use a \\xHH escape"
-                    ),
-                    format!("\\x{:X}", c as u32),
-                    Applicability::MaybeIncorrect,
-                );
-            } else if mode == Mode::Byte {
+            if mode == Mode::Byte {
```

Comment:
Is it true that we can only reach this branch for multibyte characters? My limited knowledge of utf8 would suggest we do:
```rust
if mode != Mode::RawByteStr {
let mut utf8 = String::new();
utf8.push(c);
if mode == Mode::Byte && utf8.as_bytes().count() > 1 {
    // error "this is multibyte"
} else {
    // suggest \xHH
}
```